gnat2goto/driver/tree_walk.adb

```diff
@@ -4722,124 +4718,6 @@ package body Tree_Walk is
-                                 A_Symbol_Table  => Global_Symbol_Table,
-                                 Source_Location => Source_Loc);
-
-         Func_Symbol : Symbol;
-         Array_Copy : constant Irep :=
-           Fresh_Var_Symbol_Expr (Ptr_Type, "new_array");
-         Body_Block : constant Irep := Make_Code_Block (Source_Loc);
-         Call_Args : constant Irep := New_Irep (I_Argument_List);
-         Lhs_fun_call : constant Irep :=
-           Fresh_Var_Symbol_Expr (Do_Type_Reference (Element_Type),
-                                  "array_dup_fun_lhs");
-         Call_Inst : constant Irep :=
-           Make_Code_Function_Call (Arguments       => Call_Args,
-                                    --  the argument are appended later
-                                    I_Function      =>
-                                      Get_Array_Copy_Function (
-                                        Element_Type,
-                                        Element_Type,
-                                        Index_Type),
-                                    Lhs             => Lhs_fun_call,
-                                    Source_Location => Source_Loc);
-         Return_Inst : constant Irep :=
-           Make_Code_Return (Return_Value    => Array_Copy,
-                             Source_Location => Source_Loc);
-
-         --  new variables for malloc allocation
-         Member_Size : constant Irep :=
-           Make_Constant_Expr (Source_Location => Source_Loc,
-                               I_Type          => Len_Type,
-                               Range_Check     => False,
-                               Value           =>
-                    Convert_Uint_To_Hex (Value     => Esize (Element_Type) / 8,
-                                         Bit_Width => 64));
-      begin
-         --  Create body (allocate and then call array_copy)
-         Append_Declare_And_Init (
-           Array_Copy,
-           Make_Op_Typecast (
-             Op0 =>  Make_Malloc_Function_Call_Expr (
-               Make_Op_Mul (
-                 Lhs => Member_Size,
-                 Rhs => Param_Symbol (Len_Param),
-                 I_Type => Len_Type,
-                 Source_Location => Source_Loc)),
-                 I_Type => Ptr_Type,
-             Source_Location => Source_Loc),
-                                  Body_Block, 0);
-
-         Append_Argument (Call_Args, Array_Copy);
-         Append_Argument (Call_Args, Param_Symbol (Ptr_Param));
-         Append_Argument (Call_Args, Param_Symbol (Len_Param));
-         Append_Op (Body_Block, Call_Inst);
-         Append_Op (Body_Block, Return_Inst);
-
-         Func_Symbol :=
-           New_Function_Symbol_Entry (Name           => Func_Name,
-                                      Symbol_Type    => Func_Type,
-                                      Value          => Body_Block,
-                                      A_Symbol_Table => Global_Symbol_Table);
-
```

Comment:
Glad to see this gone. 
Less happy that we never resolved the original issue why the "let_expr" and parameter passing failed.